docs/concepts/architecture-reference/azure-functions.rst

```diff
@@ -0,0 +1,332 @@
+	      "timestamp" TIMESTAMP WITHOUT TIME ZONE
+	   ),
+	   "debug" OBJECT(DYNAMIC)
+	) CLUSTERED INTO 4 SHARDS
+	  PARTITIONED BY ("ts_month");
+
+Azure Function
+--------------
+
+We should first create an environment for our function application. Following the documentation
+for whichever environment we would develop in (for example, `Visual Studio Code`_ ), we should have
+a folder structure like this::
+
+	ReferenceArchitectureProject
+	 | - IoTHubToCrateDBFunction
+	 | | - index.js
+	 | | - function.json
+	 | - node_modules
+	 | - host.json
+	 | - package.json
+	 | - extensions.csproj
+	 | - local.settings.json
+
+Our azure function, ``IoTHubToCrateDBFunction``, will be composed of 2 files:
+
+	- ``function.json`` which defines the metadata of our function.
+	- ``index.js`` which is our core Azure Function.
+
+
+Dependencies
+............
+
+Our Azure Function will rely on the `node-postgres`_ library to connect to CrateDB.
+We can install this dependency by executing::
+
+	$ npm install pg
+
+within the ``ReferenceArchitectureProject`` folder. The dependency should now be in the ``node_modules`` folder,
+which we will bundle with our azure function when we deploy the application.
+
+
+``function.json``
+.................
+
+Our ``function.json`` is fairly straightforward, and just defines what events
+we want to bind to, as well as things like eventhub names, consumer groups, etc.
+
+.. code-block:: json
+
+	{
+	  "bindings": [{
+	    "type": "eventHubTrigger",
+	    "name": "eventHubMessages",
+	    "direction": "in",
+	    "eventHubName": "default",
+	    "connection": "EventHubConnectionString",
+	    "cardinality": "many",
+	    "consumerGroup": "cratefunction"
+	  }]
+	}
+
+The parameters here can benefit from some elucidation.
+
+	- ``type`` signifies the type of binding we associate with this function. In this case, it is an
+	  eventhub trigger.
+	- ``name`` signifies the name that is used for the bound data in the Azure Function.
+	- ``direction`` signifies whether the binding is for recieving data into the function, or sending the
+	  data from the function. We have written ``in`` as the function is concerned with reading data in from
+	  the eventhub.
+	- ``eventHubName`` signifies the name of the event hub. The event hub name value in the connection string
+	  overwrites this paramter at run time, if it is set.
+	- ``connection`` signifies the name of the Azure Function application setting that contains our event hub's
+	  namespace connection string.
+	- ``cardinality`` signifies whether or not we want to use batching, or to consume messages one by one. Batching
+	  is far more efficient for our use case, which is why it is set to ``many`` instead of ``one``.
+	- ``consumerGroup`` signifies which consumer group will be used to consume messages from the eventhub. Ensure
+	  that it is unique, and not already being used by another consumer. If omitted, it will use the default consumer
+	  group of ``$Default``.
```

Comment:
consider turning this into a table for easy parsing and removing redundancies such as 'signifies' 

recieving = receiving

run time = runtime